src/udev/udev-builtin-input_id.c

```diff
@@ -44,6 +44,12 @@
 #define LONG(x) ((x)/BITS_PER_LONG)
 #define test_bit(bit, array)    ((array[LONG(bit)] >> OFF(bit)) & 1)
 
+/* key code ranges above BTN_MISC (start is inclusive, stop is exclusive)*/
+static const unsigned high_key_blocks[] = {
+        KEY_OK, BTN_DPAD_UP,
+        KEY_ALS_TOGGLE, BTN_TRIGGER_HAPPY
+};
```

Comment:
This and the loop below would be more elegant/simpler as a list of struct. But just thinking aloud, it's good enough as-is IMHO.